astropy/units/tests/test_logarithmic.py

```diff
@@ -196,18 +196,24 @@ def test_str(self):
 
         lu4 = u.mag(u.ct)
         assert lu4.to_string("generic") == "mag(ct)"
-        latex_str = r"$\mathrm{mag}$$\mathrm{\left( \mathrm{ct} \right)}$"
+        latex_str = r"$\mathrm{mag\left(ct\right)}$"
         assert lu4.to_string("latex") == latex_str
         assert lu4.to_string("latex_inline") == latex_str
         assert lu4._repr_latex_() == latex_str
 
         lu5 = u.mag(u.ct / u.s)
-        assert lu5.to_string("latex") == (
-            r"$\mathrm{mag}$$\mathrm{\left( \mathrm{\frac{ct}{s}} \right)}$"
-        )
-        latex_str = r"$\mathrm{mag}$$\mathrm{\left( \mathrm{ct\,s^{-1}} \right)}$"
+        latex_str = r"$\mathrm{mag\left(\frac{ct}{s}\right)}$"
+        assert lu5.to_string("latex") == latex_str
+        latex_str = r"$\mathrm{mag\left(ct\,s^{-1}\right)}$"
         assert lu5.to_string("latex_inline") == latex_str
 
+    def test_dex_latex_str(self):
+        # Regression test for gh-18618.
+        lu = u.dex(u.cm / u.s**2)
+        latex_str = r"$\mathrm{dex\left(\frac{cm}{s^{2}}\right)}$"
```

Comment:
@Mhvk shouldn't this have been?

```suggestion
        latex_str = r"$\mathrm{dex}\left(\frac{cm}{s^{2}}\right)$"
```

Reply:
No, I think it is correct: all the text in units should be typeset rm. I remove the `\mathrm` for the physical unit in the construction of this.

Reply:
Darn, I should have tested the full units suite, though!